### DISCUSSION TITLE
How do I create CLI script that is able to use `'$env/static/private`?

### DISCUSSION THREAD

**Discussion by mrceperka** [sveltejs/kit#10812] [Q&A] [answered]
I have a script `src/lib/db/sync.ts`. 
```ts
// getRaids uses $env/static/private
import { getRaids } from '../api';
import { db } from './index';
import { bossTable } from './schema/boss.schema';
import { raidTable } from './schema/raid.schema';

const main = async () => {
	const raids = await getRaids();
	for (const raid of raids) {
		await db
			.insert(raidTable)
			.values([{ name: raid.map }])
			.execute();
		for (const boss of raid.bosses) {
			await db.insert(bossTable).values({ entry: boss.entry, name: boss.name }).execute();
		}
	}
};
main();

```
I run this script with like this `node --loader esbuild-register/loader -r esbuild-register --no-warnings ./src/lib/server/db/synchronize-with-api.ts` but I got this error

```
Error: Cannot find module '$env/static/private'
```

Is there any way around this? I just want to run simple CLI script

**Comment by kevbook** [comment]
You probably need to use `vite` to run the script

**Reply by mrceperka** [reply]
Yeah, but I did not now how exactly one can do it :D

**Comment by danielres** [comment, marked as answer]
I've used `vite-node` successfully in the past.
https://github.com/vitest-dev/vitest/tree/main/packages/vite-node#readme

**Reply by danielres** [reply]
Here for example: https://github.com/danielres/pixel-avatar-editor/blob/v2-save-and-dl/package.json#L16

**Reply by mrceperka** [reply]
I will give it a try, this sounds very good. Thanks :pray:

**Reply by mrceperka** [reply]
@danielres any idea how do I convince sveltekit that I am runnig this on server? `vite-node script-that-import-from-private-env.ts` gives me this error:
```
Error: Cannot import $env/static/private into client-side code
```

I was able to overcome this with `TEST=true` env variable (https://kit.svelte.dev/docs/server-only-modules#further-reading) but I do not like it at all :D

**Reply by danielres** [reply]
if you want access an environment variable on the frontend side, you have to:

1. prefix its name with PUBLIC_
2. access it through `$env/static/public`

This is to ensure you don't leak any sensitive env var to the frontend accidentally.

**Reply by david-plugge** [reply]
https://github.com/sveltejs/kit/discussions/10985#discussioncomment-7495285
